### ISSUE TITLE
Use setuptools_scm for version management

### ISSUE THREAD

**Issue by diazona** [pytest-dev/pytest-localserver#8]
This is just a proposal, not something we necessarily have to do, but I wanted to get it written down before I forget: I've used [setuptools_scm](https://github.com/pypa/setuptools_scm/) for automatic version management in many of my other projects and I really like the convenience of not having to update the version number manually in files. Should we adopt it here as well?

**Comment by redtoad** [comment]
I'm in two minds about this. It feels a bit over the top when we only have two places where the version number appears: in `setup.py` and in `CHANGES`. However, I have zero experience with setuptools_scm.

**Comment by RonnyPfannschmidt** [comment]
Im it's maintainer and we use it as part of the release automation in many projects

It's all about having a single source of truth and preventing human error

**Comment by diazona** [comment]
Oh hey small world, well thank you for all your work on it @RonnyPfannschmidt :smiley:

Anyway, @redtoad adopting `setuptools_scm` would mean we'd no longer record the version number in `setup.py`, it would be recorded as a Git tag instead. Other than that, there's really no need to think about it. So it's roughly the same amount of work. But I think it's handy to have tags that correspond to releases and we'd get that for free. Plus, we'd get the benefit that every single commit automatically gets its own unique version number which incorporates the commit hash, which is not really a big deal but I've found it to come in handy for preparing one-off packages of unreleased versions for testing.

And to be clear, I'm not suggesting this for the upcoming 0.5.1 release; I thought if we want to make this change, the time to do it would be after that, maybe for version 0.6. I just posted the issue because I didn't want to forget about it.
